### PR TITLE
12310 Nature Of Business tweaks

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#12310

*Description of changes:*
- updated code-desc display
- updated state conditions
- updated button right alignment
- updated indents (lint)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/nature-of-business/NatureOfBusiness.vue [hunk 1]
```diff
@@ -180,24 +184,16 @@ export default class NatureOfBusiness extends Vue {
 
   /** Whether we have stored NAICS data. */
   get haveNaics (): boolean {
-    return (!!this.naicsCode && !!this.naicsDescription)
+    return (!!this.naicsCode || !!this.naicsDescription)
```
Comment (severinbeauvais) [inline src/components/nature-of-business/NatureOfBusiness.vue:187]: This is to handle the situation where we have a code and no description (or vice-versa).

(Yes, this is now a valid scenario.)

File: src/components/nature-of-business/NatureOfBusiness.vue [hunk 2]
```diff
@@ -180,24 +184,16 @@ export default class NatureOfBusiness extends Vue {
...
     return (this.haveNaics && this.state === States.SUMMARY)
   }
 
-  /** Called when this component has been created. */
-  created (): void {
-    // on init, if we have stored NAICS data then display summary
-    if (this.haveNaics) {
-      this.state = States.SUMMARY
-    }
-  }
```
Comment (severinbeauvais) [inline src/components/nature-of-business/NatureOfBusiness.vue:197]: The problem with this is the state is only updated when the component is created -- not when the props change.

Now we use a watcher instead (see below).

File: src/components/nature-of-business/NatureOfBusiness.vue [hunk 3]
```diff
@@ -285,7 +287,12 @@ export default class NatureOfBusiness extends Vue {
   padding: 8px;
 }
 
+#nob-change-btn {
+  margin-right: -14px;
```
Comment (severinbeauvais) [inline src/components/nature-of-business/NatureOfBusiness.vue:291]: This is so the button label is in line with the right margin, but when highlight the padding is the same on the left and right (it goes past the margin), ie:

image

Comment (cameron-freshworks) [reply src/components/nature-of-business/NatureOfBusiness.vue:291]: I think* the Edit UI might already have some compensation for this, so might need an update there, super minor 👍

Comment (severinbeauvais) [reply src/components/nature-of-business/NatureOfBusiness.vue:291]: It does. I'm in there already so I removed the v-deep.

**Pull request merged**